modules/openy_gc_auth/modules/openy_gc_auth_custom/src/Form/VirtualYCustomLoginForm.php

```diff
@@ -166,6 +184,21 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
       ]));
       $this->flood->register('openy_gc_auth_custom.login', $flood_config->get('user_window'));
     }
+
+    // Restrict access to user 1 and users with roles except virtual_y*.
+    $user = reset($users);
+    $account_roles = $user->getRoles();
+    // Remove all virtual_y roles.
+    foreach ($account_roles as $id => $account_role) {
+      if (strstr($account_role, 'virtual_y') !== FALSE || $account_role == 'authenticated') {
+        unset($account_roles[$id]);
+      }
+    }
+    if ($user->id() == 1 || !empty($account_roles)) {
+      $form_state->setErrorByName('email', $this->t('This user is not allowed to login from this form.', [
```

Comment:
@hamrant would it make sense to direct users to the proper login form here?

Reply:
Let's leave the origin form hidden for now